### PR TITLE
document path-to-regexp change in express 5 migration guide

### PR DESCRIPTION
**Pull request by HarshithaKP**

### CODE REVIEW

**Comment by HarshithaKP** [comment]
Pre-req: https://github.com/pillarjs/router/pull/42
Refs: https://github.com/expressjs/discussions/issues/100#issuecomment-590058884

File: en/guide/migrating-5.md [hunk 1]
```diff
@@ -116,6 +117,14 @@ The `res.sendfile()` function has been replaced by a camel-cased version `res.se
 
 The `app.router` object, which was removed in Express 4, has made a comeback in Express 5. In the new version, this object is a just a reference to the base Express router, unlike in Express 3, where an app had to explicitly load it.
 
+<h4 id="path-to-regexp"><path-to-regexp</h4>
+
+From Express 5 onwards, following are the changes with respect to path-to-regexp which has direct implications to route.
+* RegExp special characters can now only be used in a parameter, as opposed to anywhere.
+* Parameters can have suffixes to augment meaning - *, + and ?. E.g. /:user*
```
Comment (blakeembrey) [inline en/guide/migrating-5.md]: Looks like you need to escape or use code for the markdown special characters, they're currently italicizing the text.

Comment (HarshithaKP) [reply en/guide/migrating-5.md]: @blakeembrey, thanks. Used code for special characters. PTAL.

**Comment by gireeshpunathil** [comment]
one approval, and this is should be ready to go in!
/cc @blakeembrey @dougwilson @wesleytodd @LinusU

**Review by LinusU — APPROVED** [review]
LGTM 👍 

Potentially we could also add explanations for the different suffixes?

**Comment by gireeshpunathil** [comment]
> Potentially we could also add explanations for the different suffixes?

@LinusU - from your approval, I am assuming that this comment is not a blocker?

@expressjs/express-tc - this is all set to land, towards express 5 GA. Is there anyone with spare bandwidth to review and merge this?

File: en/guide/migrating-5.md [hunk 2]
```diff
@@ -116,6 +117,14 @@ The `res.sendfile()` function has been replaced by a camel-cased version `res.se
 
 The `app.router` object, which was removed in Express 4, has made a comeback in Express 5. In the new version, this object is a just a reference to the base Express router, unlike in Express 3, where an app had to explicitly load it.
 
+<h4 id="path-to-regexp"><path-to-regexp</h4>
+
+From Express 5 onwards, following are the changes with respect to path-to-regexp which has direct implications to route.
+* RegExp special characters can now only be used in a parameter, as opposed to anywhere.
+* Parameters can have suffixes to augment meaning -`*`, `+` and `?`. E.g. `/:user*`
```
Comment (jonchurch) [inline en/guide/migrating-5.md]:
```suggestion
* Parameters can have suffixes to augment meaning and affect how matching works:
 	* `*`: [Zero or more](https://github.com/pillarjs/path-to-regexp#zero-or-more), e.g. `/:foo*` matches `/bar/baz` and `/`
 	* `+`: [One or more](https://github.com/pillarjs/path-to-regexp#one-or-more), e.g. `/:foo+` matches `/bar` and `/bar/baz` but not `/`
 	* `?`: [Optional](https://github.com/pillarjs/path-to-regexp#optional), e.g. `/:foo/:bar?` matches `/bar` and `/bar/baz`
```
Expanded on this explanation a bit by pulling in examples from path-to-regexp

**Comment by LinusU** [comment]
> from your approval, I am assuming that this comment is not a blocker?

Correct 👍 

Although @jonchurch have now provided an extend description which I think we should use 👏

File: en/guide/migrating-5.md (hunk 2 above)
Comment (HarshithaKP) [reply en/guide/migrating-5.md]: @jonchurch, thanks. Included your suggestion. PTAL.

**Review by jonchurch — APPROVED** [review]

**Comment by dougwilson** [comment]
I believe we need to wait for the express release with this change to be published prior to landing on the site (i.e. 5.0.0-beta.1)

**Pull request merged**